internal/runner/jest_test.go

```diff
@@ -112,6 +112,35 @@ func TestJestRun_Retry(t *testing.T) {
 	}
 }
 
+func TestJestRun_Retry_WithTestFilePattern(t *testing.T) {
+	changeCwd(t, "./testdata/jest")
+
+	jest := NewJest(RunnerConfig{
+		TestCommand:      "jest --invalid-option --json --outputFile {{resultPath}}",
+		RetryTestCommand: "npx jest {{testExamples}} --testNamePattern '{{testNamePattern}}' --json --outputFile {{resultPath}}",
+		ResultPath:       "jest.json",
+	})
+
+	t.Cleanup(func() {
+		os.Remove(jest.ResultPath)
+	})
+
+	testCases := []plan.TestCase{
+		{Scope: "expelliarmus", Name: "disarms the opponent", Path: "./testdata/jest/spells/expelliarmus.spec.js:1:1"},
+		{Scope: "lumos", Name: "illuminates the room", Path: "./testdata/jest/spells/lumos.spec.js:1:1"},
+	}
+	result := NewRunResult([]plan.TestCase{})
+	err := jest.Run(result, testCases, true)
+
+	if err != nil {
+		t.Errorf("Jest.Run(%q) error = %v", testCases, err)
+	}
+
+	if result.Status() != RunStatusPassed {
+		t.Errorf("Jest.Run(%q) RunResult.Status = %v, want %v", testCases, result.Status(), RunStatusPassed)
+	}
```

Comment:
[blocking] We probably want to add an assertion to check test result count to equal `2` to prove that the file name filter works. 

(I presume this is the reason why you added a new test fixture?)